northstar-runtime/src/runtime/fork/init/mod.rs

```diff
@@ -141,47 +140,34 @@ impl Init {
-                    let stdin = io[0];
-                    let stdout = io[1];
-                    let stderr = io[2];
-
                     // Start new process inside the container
-                    let pid = fork(|| {
-                        util::set_parent_death_signal(Signal::SIGKILL);
-
-                        unistd::dup2(stdin, nix::libc::STDIN_FILENO).expect("failed to dup2");
-                        unistd::dup2(stdout, nix::libc::STDOUT_FILENO).expect("failed to dup2");
-                        unistd::dup2(stderr, nix::libc::STDERR_FILENO).expect("failed to dup2");
+                    let pid = match unsafe { fork().expect("failed to fork") } {
+                        ForkResult::Parent { child } => child.as_raw() as Pid,
+                        ForkResult::Child => {
+                            util::set_parent_death_signal(Signal::SIGKILL);
+
+                            // Set seccomp filter
+                            if let Some(ref filter) = self.seccomp {
+                                filter.apply().expect("failed to apply seccomp filter.");
+                            }
 
-                        unistd::close(stdin).expect("failed to close stdout after dup2");
-                        unistd::close(stdout).expect("failed to close stdout after dup2");
-                        unistd::close(stderr).expect("failed to close stderr after dup2");
+                            let path = CString::from(path);
+                            let args: Vec<_> = args.into_iter().map_into::<CString>().collect();
+                            let env: Vec<_> = env.into_iter().map_into::<CString>().collect();
 
-                        // Set seccomp filter
-                        if let Some(ref filter) = self.seccomp {
-                            filter.apply().expect("failed to apply seccomp filter.");
+                            panic!(
+                                "execve: {:?} {:?}: {:?}",
+                                &path,
+                                &args,
+                                unistd::execve(&path, &args, &env)
+                            )
                         }
+                    };
 
-                        let path = CString::from(path);
-                        let args = args.into_iter().map_into::<CString>().collect_vec();
-                        let env = env.into_iter().map_into::<CString>().collect_vec();
-
-                        panic!(
-                            "execve: {:?} {:?}: {:?}",
-                            &path,
-                            &args,
-                            unistd::execve(&path, &args, &env)
-                        )
-                    })
-                    .expect("failed to spawn child process");
-
-                    // close fds
+                    // Close the console fd used in the container binary only.
                     drop(console);
-                    unistd::close(stdin).expect("failed to close stdout");
```

Comment:
Why did we explicitly close those `OwnedFd`s if their `Drop` implementation does it for us?

Reply:
In this code the `stdin` `stdout` and `stderr` are `RawFd` and need to be closed. They're received here:

```rust
let io = stream.recv_fds::<RawFd, 3>().expect("failed to receive io");
```